### DISCUSSION TITLE
Else statement

### DISCUSSION THREAD

**Discussion by lobre** [maragudk/gomponents#131] [Q&A] [answered]
Hi,

In `html/template`, you can use `if`/`else` instructions. In gomponents, only `.If` is supported.

Here is an example.

```
{{if .IsAuthenticated}}
  Hello {{.Username}}!
{{else}}
  Please log in.
{{end}}
```

How would to handle such a situation with gomponents? Could a `.If(...).Else(...)` be useful?

Same but different, how would you handle a loop on a range?

Thanks

**Comment by markuswustenberg** [comment, marked as answer]
Basically, negate the conditional expression:

```go
func component(condition bool) g.Node {
  return Div(
    g.If(condition, …),
    g.If(!condition, …),
  )
}
```

Instead of looping, use [Map](https://pkg.go.dev/github.com/maragudk/gomponents#Map). Or is there something you need that you don't get from looping?

**Reply by markuswustenberg** [reply]
By the way, you can always do something like this if you want imperative code inline:

```go
func component(numbers []int) g.Node {
  return Div(
    func() g.Node {
      for _, n := range numbers {
        …
      }
      return someOtherComponent(…)
    }()
  )
}
```

**Comment by lobre** [comment]
Great examples, thanks for providing those. They answer my concerns!

**Comment by wworrall** [comment]
Thank you @markuswustenberg for providing a simple solution to this. I came across this in a search for If/Else functionality too. I often find myself writing:

```
g.If(condition, ...)
g.If(!condition, ...)
```

Whilst we could all add a simply utility function to our projects, I wonder whether it is such a common need that it might be worth adding to gomponents?

A possible function signature:

```
g.IfElse(
   condition,
   trueComponent(),
   falseComponent(),
)
```

Happy to contribute but unsure whether this aligns with the ethos of the lib? I, personally, find myself wanting this in gomponents.

**Reply by markuswustenberg** [reply]
I actually really like the simplicity of just having `If`, but I acknowledge that it comes down to code style. Personally, I almost never use `else` anywhere unless it's really the most obvious solution.

So I'm leaning towards not adding it. I favor having a smaller API vs. the arguably small added convenience of not having to negate the conditional expression.

**Reply by wworrall** [reply]
Sure, I appreciate this! And on consideration, if the lib were to support if/else, then it would lead to if/else if/else support which perhaps would require chaining. And the complexity grows.
